core/src/test/java/org/neverplan2/wololo/service/ScannerServiceTest.java

```diff
@@ -1,68 +1,48 @@
 package org.neverplan2.wololo.service;
 
-import ch.qos.logback.classic.Logger;
-import ch.qos.logback.classic.spi.ILoggingEvent;
-import ch.qos.logback.core.read.ListAppender;
-import org.junit.AfterClass;
-import org.junit.Before;
-import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
-import org.neverplan2.wololo.net.exception.NetworkScannerException;
-import org.slf4j.LoggerFactory;
+import org.neverplan2.wololo.api.model.NetAddress;
+import org.neverplan2.wololo.api.model.NicAdapter;
+import org.neverplan2.wololo.dao.NetAddressEntity;
+import org.neverplan2.wololo.dao.NicAdapterEntity;
+import org.neverplan2.wololo.repo.NicAdapterRepository;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertTrue;
+import java.util.Arrays;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.when;
 
 @RunWith(MockitoJUnitRunner.class)
 public class ScannerServiceTest {
 
     @InjectMocks
     protected ScannerService scannerService;
 
-    private static ListAppender<ILoggingEvent> listAppender;
-
-    @BeforeClass
-    public static void prepareSet() {
-        Logger root = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
-        listAppender = new ListAppender<>();
-        listAppender.start();
-        root.addAppender(listAppender);
-    }
+    @Mock
+    private NicAdapterRepository nicAdapterRepository;
 
-    @Before
-    public void prepareTest() {
-        scannerService.init();
-    }
 
     @Test
-    public void initTest() {
-        this.scannerService.init();
-        //assertEquals("No scanner registered.", listAppender.list.get(0).getMessage());
-        assertTrue(true);
+    public void scanNicAdapters() {
+        when(nicAdapterRepository.findAll()).thenReturn(Arrays.asList(new NicAdapterEntity(), new NicAdapterEntity()));
+        List<NicAdapter> newAddress = scannerService.scanNicAdapters();
+        assertEquals(2, newAddress.size());
     }
 
     @Test
-    public void scanners() {
-        assertEquals(1, scannerService.getNetworkScanners().size());
+    public void scanNetwork() {
```

Comment:
best we do not create empty tests so coverage will always remind us that we still have work to do